server/src/core/user/user-service.ts

```diff
@@ -465,6 +469,20 @@ export class Users implements UserService {
     }
   }
 
+  public async getByPhoneOrEmail(phone: string, email: string): Promise<User> {
+    try {
+      const user = await this.collection.findOne({ $or: [ { phone }, { email } ] }, { projection: SAFE_USER_PROJECTION });
+      if (user) {
+        return getSafeUser(user);
+      }
```

Comment:
I think the `projection: SAFE_USER_PROJECTION` already handles this so that you don't have to call `getSafeUser` separately.